server/webapp/WEB-INF/rails/webpack/views/pages/pipeline_configs/spec/stages/settings/tabs/settings_spec.tsx

```diff
@@ -0,0 +1,155 @@
+  });
+
+  it("should render approval type switch", () => {
+    expect(helper.findByDataTestId("switch-wrapper")).toBeInDOM();
+    expect(helper.findByDataTestId("switch-label")).toContainText("Automatically run this stage on upstream changes:");
+    expect((helper.findByDataTestId("switch-checkbox")[0] as HTMLInputElement).checked).toEqual(true);
+  });
+
+  it("should allow toggling approval switch", () => {
+    expect(helper.findByDataTestId("switch-wrapper")).toBeInDOM();
+
+    expect(stageConfig.approval().state()).toEqual(true);
+    expect((helper.findByDataTestId("switch-checkbox")[0] as HTMLInputElement).checked).toEqual(true);
+
+    helper.findByDataTestId("switch-checkbox").val("off");
+    simulateEvent.simulate(helper.findByDataTestId("switch-checkbox")[0], "click");
+    m.redraw.sync();
+
+    expect(stageConfig.approval().state()).toEqual(false);
+    expect((helper.findByDataTestId("switch-checkbox")[0] as HTMLInputElement).checked).toEqual(false);
+  });
+
+  it("should render fetch materials checkbox", () => {
+    expect(helper.findByDataTestId("form-field-input-fetch-materials")).toBeInDOM();
+    expect(helper.findByDataTestId("form-field-label-fetch-materials")).toBeInDOM();
+
+    expect((helper.findByDataTestId("form-field-input-fetch-materials")[0] as HTMLInputElement).checked).toEqual(false);
+    expect(helper.findByDataTestId("form-field-label-fetch-materials")).toContainText("Fetch Materials");
+  });
+
+  it("should allow toggling fetch materials checkbox", () => {
+    expect((helper.findByDataTestId("form-field-input-fetch-materials")[0] as HTMLInputElement).checked).toEqual(false);
+
+    helper.findByDataTestId("form-field-input-fetch-materials").val("on");
+    simulateEvent.simulate(helper.findByDataTestId("form-field-input-fetch-materials")[0], "click");
+
+    expect((helper.findByDataTestId("form-field-input-fetch-materials")[0] as HTMLInputElement).checked).toEqual(true);
+  });
+
+  it("should render fetch materials checkbox", () => {
+    expect(helper.findByDataTestId("form-field-input-fetch-materials")).toBeInDOM();
+    expect(helper.findByDataTestId("form-field-label-fetch-materials")).toBeInDOM();
+
+    expect((helper.findByDataTestId("form-field-input-fetch-materials")[0] as HTMLInputElement).checked).toEqual(false);
+    expect(helper.findByDataTestId("form-field-label-fetch-materials")).toContainText("Fetch Materials");
+  });
+
+  it("should allow toggling fetch materials checkbox", () => {
+    expect((helper.findByDataTestId("form-field-input-fetch-materials")[0] as HTMLInputElement).checked).toEqual(false);
+
+    helper.findByDataTestId("form-field-input-fetch-materials").val("on");
+    simulateEvent.simulate(helper.findByDataTestId("form-field-input-fetch-materials")[0], "click");
+
+    expect((helper.findByDataTestId("form-field-input-fetch-materials")[0] as HTMLInputElement).checked).toEqual(true);
+  });
```

Comment:
We need to remove this, as it is duplicate :)

Reply:
Not sure if I understood the duplicate code.
Can you point the duplicate piece of code? line?

Reply:
https://github.com/gocd/gocd/pull/6814/files#diff-bd2007565d8172520b0dfe850e2ba3beR104-R119

Reply:
Hmm. Not sure if I understood the duplicate part.
* `should render fetch materials checkbox` verifies:
    - `fetch-materials` checkbox is rendered on the screen.
    - `fetch-materials`  checkbox is unchecked by default.
* `should allow toggling fetch materials checkbox ` verifies:
   - Initially, the `fetch-materials` checkbox is unchecked by default. And on click, the checkbox state is toggled.

Reply:
Sorry for the wrong link

https://github.com/gocd/gocd/pull/6814/files#diff-bd2007565d8172520b0dfe850e2ba3beR87-R119

Reply:
Fixed as part of https://github.com/gocd/gocd/pull/6822/files#diff-bd2007565d8172520b0dfe850e2ba3be